src/modules/cartPage.ts

```diff
@@ -0,0 +1,134 @@
+    private onButtonClick: OnButtonClick;
+
+    constructor(
+        parent: Element,
+        productsData: IProduct[],
+        selectedProducts: number[],
+        onChangeAmount: OnChangeAmount,
+        onProductClick: OnProductClick,
+        onButtonClick: OnButtonClick
+    ) {
+        this.productsData = productsData;
+        this.parent = parent;
+        this.selectedProducts = selectedProducts;
+        this.onChangeAmount = onChangeAmount;
+        this.onProductClick = onProductClick;
+        this.onButtonClick = onButtonClick;
+
+        this.cartDataLayout = `
+            <div class="cart-page_wrapper">
+                <div class="selected-list">
+                    <h2 class="h2">Bag</h2>
+                </div>
+                <div class="summary">
+                    <h3 class="h3 summary__header"></h3>
+                    <input class="input promo_code" type="text" placeholder="Promo code">
+                    <p class="summary__total-price"></p>
+                    <button type="submit" class="button button_buy">Buy</button>
+                </div>
+            </div>
+            <div class="disabled-area hidden"></div>`;
+    }
+
+    public drawCartPage() {
+        this.parent.innerHTML = this.cartDataLayout;
+        const selectedList: Element = document.querySelector('.selected-list')!;
+
+        const summaryHeader: Element = document.querySelector('.summary__header')!;
+        const totalPriceContainer: Element = document.querySelector('.summary__total-price')!;
+        const inputPromoCode = document.querySelector('.promo_code')! as HTMLInputElement;
+        let totalAmount: number = 0;
+        let totalPrice: number = 0;
+        this.productsData.forEach((product) => {
+            if (this.selectedProducts.includes(product.id)) {
+                let amount = this.selectedProducts.reduce((amount, currentItem) => {
+                    if (currentItem === product.id) {
+                        amount += 1;
+                    }
+                    return amount;
+                }, 0);
+                totalAmount += amount;
+                totalPrice += amount * product.price;
+                selectedList.appendChild(this.drawCartItem(product, amount));
+            }
+        });
+        inputPromoCode.addEventListener('input', () => {
+            totalPrice = this.checkPromo(inputPromoCode, totalPrice, totalPriceContainer);
+        });
+        summaryHeader.innerHTML = `<span>Summary</span> <span>[${totalAmount} items]</span>`;
+        totalPrice = this.checkPromo(inputPromoCode, totalPrice, totalPriceContainer);
+        totalPriceContainer.innerHTML = `<span>Total</span> <span>€‎${totalPrice}</span>`;
+
+        const buyButton: Element = document.querySelector('.button_buy')! as HTMLButtonElement;
+        buyButton.addEventListener('click', () => this.onButtonClick('buy'));
+    }
+
+    private drawCartItem(itemData: IProduct, amount: number): Element {
```

Comment:
This is quite large method. 
Consider splitting into smaller meaningful sections.

Reply:
Fixed.